gax-grpc/src/main/java/com/google/api/gax/grpc/OperationCallable.java

```diff
@@ -173,9 +184,65 @@ public ResponseT call(RequestT request) {
         operationsClient
             .getOperationCallable()
             .futureCall(GetOperationRequest.newBuilder().setName(operationName).build());
-    OperationFuture<ResponseT> operationFuture =
-        OperationFuture.create(
-            operationsClient, getOperationFuture, clientContext.getExecutor(), responseClass);
-    return operationFuture;
+    return futureCall(getOperationFuture);
+  }
+
+  /**
+   * Sends a cancellation request to the server for the operation with name {@code operationName}.
+   *
+   * @param operationName The name of the operation to cancel.
+   * @return the future which completes once the operation is canceled on the server side.
+   */
+  public ApiFuture<Empty> cancel(String operationName) {
```

Comment:
OperationFuture was the only heavy future in the library. Basically future is just an object a client can use to wait for completion of something. The actual logic is usually somewhere else. I would like to keep future as simple as possible.The first line of documentation of `java.util.concurrent.Future` says the following:

A Future represents the result of an asynchronous computation.

So a future represents a result of computation, not the computation itself.

Also, for this particular case, cancel() of the server-side operation will conflict with cancel() of the future itself, which are two different things.